client/core/core.go

```diff
@@ -396,6 +396,19 @@ func (dc *dexConnection) hasActiveOrders() bool {
 	return false
 }
 
+// maybeClearWalletData will call the ClearData method if the wallet implements
+// DataClearer and does not currently have any active orders.
+func (c *Core) maybeClearWalletData(wallet *xcWallet) {
+	clearer, isClearer := wallet.Wallet.(asset.DataClearer)
+	if !isClearer {
+		return
+	}
+
+	if !c.assetHasActiveOrders(wallet.AssetID) {
+		clearer.ClearData()
```

Comment:
I had this doubly linked at first, but I changed it to the one time garbage collection at start up because I thought it would be safe to assume that people generally wouldn't shut the `dexc` down while having active deals. I guess you're right, some market makers might never turn it off.